src/plone/restapi/tests/test_serializer_catalog.py

```diff
@@ -31,7 +31,7 @@ def setUp(self):
         # /plone/my-folder/my-document
         self.doc = createContentInContainer(
             self.folder, u'Document',
-            created=DateTime(2015, 12, 31, 23, 45),
+            creation_date=DateTime(2015, 12, 31, 23, 45),
```

Comment:
This requires an explanation:

In Dexterity the attribute name that saves the creation date is not 'created' but 'creation_date'. https://github.com/plone/plone.dexterity/blob/2.2.x/plone/dexterity/content.py#L508

If we leave the document creation code as is (with created), when the object is serialized to JSON in the test that uses fullobjects, we get an error saying that 'DateTime object is not callable.

```Error in test test_lazy_map_serialization_with_fullobjects (plone.restapi.tests.test_serializer_catalog.TestCatalogSerializers)
Traceback (most recent call last):
  File "/home/erral/buildout.python/parts/opt/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/erral/dev/plone.restapi/src/plone/restapi/tests/test_serializer_catalog.py", line 64, in test_lazy_map_serialization_with_fullobjects
    results = getMultiAdapter((lazy_map, self.request), ISerializeToJson)(fullobjects=True)
  File "/home/erral/dev/plone.restapi/src/plone/restapi/serializer/catalog.py", line 88, in __call__
    include_items=False)
  File "/home/erral/dev/plone.restapi/src/plone/restapi/serializer/dxcontent.py", line 58, in __call__
    'created': json_compatible(obj.created()),
TypeError: 'DateTime' object is not callable
```

Reply:
The datetime field story in Dexterity objects is quite a nightmare: there is an [effective_date and effective method on the base class](https://github.com/plone/plone.dexterity/blob/2.2.x/plone/dexterity/content.py#L516) but there is also an [effective field in the IPublication behavior](https://github.com/plone/plone.app.dexterity/blob/2.3.x/plone/app/dexterity/behaviors/metadata.py#L146), similar things happen with expiration_date and expires.